lib/features/family/features/scan_nfc/nfc_scan_screen.dart

```diff
@@ -263,13 +272,17 @@ class _NFCScanPageState extends State<NFCScanPage> {
       onClickPrimaryBtn: () async =>
           _handleGenericErrorTryAgainClicked(context),
       onClickSecondaryBtn: () {
+        if (widget.isFromGenerosityChallenge) {
```

Comment:
No i tried that at first but then you have the scan "bottomsheet" depending on android/ ios that you kinda have to re-trigger and the screen "below that". So I kinda understood why it truly goes "home" instead of back, saves some navigation hassle.